include/eve/detail/function/simd/riscv/compounds.hpp

```diff
@@ -0,0 +1,293 @@
+      self = bit_cast(i_t(intrinsic(self_cast, other_cast, N::value)), as(self));                  \
+      return self;                                                                                 \
+    }                                                                                              \
+  }                                                                                                \
+  template<plain_scalar_value T, typename N, plain_scalar_value P>                                 \
+  EVE_FORCEINLINE auto& eve_name(wide<T, N>& self, P const& other) noexcept                        \
+  requires rvv_abi<abi_t<T, N>> && (sizeof(T) == sizeof(P))                                        \
+  {                                                                                                \
+    using wide_t = wide<T, N>;                                                                     \
+    if constexpr( std::is_floating_point_v<T> )                                                    \
+    {                                                                                              \
+      using scalar_t            = as_integer_t<P, unsigned>;                                       \
+      using i_t                 = wide_t::template rebind<scalar_t, N>;                            \
+      constexpr auto tgt        = as<i_t>();                                                       \
+      auto           other_cast = std::bit_cast<scalar_t>(other);                                  \
+      auto           self_cast  = bit_cast(self, tgt);                                             \
+      self                      = bit_cast(eve_name(self_cast, other_cast), as(self));             \
+      return self;                                                                                 \
+    }                                                                                              \
+    else                                                                                           \
+    {                                                                                              \
+      using scalar_t            = as_integer_t<P, unsigned>;                                       \
+      using i_t                 = wide_t::template rebind<scalar_t, N>;                            \
+      constexpr auto tgt        = as<i_t>();                                                       \
+      auto           other_cast = std::bit_cast<scalar_t>(other);                                  \
+      auto           self_cast  = bit_cast(self, tgt);                                             \
+      self = bit_cast(i_t(intrinsic(self_cast, other_cast, N::value)), as(self));                  \
+      return self;                                                                                 \
+    }                                                                                              \
+  }
```

Comment:
I dislike macros intensely. 
I would prefer callbacks, but it's going to be annoyting because of forceinline.

For exanple we can
```
  enum class bit_op { and_, or_, xor_ }; 

  template<bit_op op, 
        plain_scalar_value T, typename N, plain_scalar_value P>                                 
  EVE_FORCEINLINE auto& rvv_bit_compound_impl(wide<T, N>& self, P const& other) {
    ...
    if constexpr (op == bit_op::and_) self_cast = __riscv_vand(self_cast, other_cast);
    if constexpr (op == bit_op::xor_) self_cast = __riscv_vxor(self_cast, other_cast);
    ...
  }

```